MHArmory.Search/OpenCL/SearchDataSerializer.cs

```diff
@@ -0,0 +1,243 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using MHArmory.Core.DataStructures;
+
+namespace MHArmory.Search.OpenCL
+{
+    public class SearchDataSerializer
+    {
+        public SerializedSearchParameters Serialize(ISolverData data)
+        {
+            var parameters = new SerializedSearchParameters();
+            
+            var heads = data.AllHeads.Where(x => x.IsSelected).Select(x => x.Equipment).ToList();
+            var chests = data.AllChests.Where(x => x.IsSelected).Select(x => x.Equipment).ToList();
+            var gloves = data.AllGloves.Where(x => x.IsSelected).Select(x => x.Equipment).ToList();
+            var waists = data.AllWaists.Where(x => x.IsSelected).Select(x => x.Equipment).ToList();
+            var legs = data.AllLegs.Where(x => x.IsSelected).Select(x => x.Equipment).ToList();
+            var charms = data.AllCharms.Where(x => x.IsSelected).Select(x => x.Equipment).ToList();
+
+            var allArmorPieces = heads.Concat(chests).Concat(gloves).Concat(waists).Concat(legs).Cast<IArmorPiece>().ToList();
```

Comment:
When / if line is too long, don't hesitate to split the statement, like:
```CS
var allArmorPieces = heads
    .Concat(chests)
    .Concat(gloves)
    .Concat(waists)
    .Concat(legs)
    .Cast<IArmorPiece>()
    .ToList();
```

The good aspect of Linq queries chaining in multi-line form is that we quickly see what is done, in which order, and it is easy to change order or comment out a line to disable a step, etc...